parsl/dataflow/dflow.py

```diff
@@ -247,6 +247,8 @@ def handle_exec_update(self, task_id, future):
              makes this callback
         """
 
+        self.tasks[task_id]['app_fu'].parent_callback(future)
```

Comment:
Why do we need this callback registration here ?

Reply:
It's not a registration - it's calling the actual callback.

The whole mechanism of update_parent registering an executor future level callback to call this app future callback is removed by this PR, and replaced by this call here (which also ultimately triggers when the exec future completes... But all the exec future completion code is now called from handle_exec_update instead of being split across two callbacks that are separately registered)

I think that AppFuture callback should go away entirely in a future PR.